src/main/java/com/cosmeticos/service/OrderService.java

```diff
@@ -225,9 +230,92 @@ public Order update(OrderRequestBody request) {
 			order.setScheduleId(orderRequest.getScheduleId());
 		}
 
+		//AQUI TRATAMOS O STATUS ACCEPTED QUE VAMOS NA SUPERPAY EFETUAR A RESERVA DO VALOR PARA PAGAMENTO
+		if (orderRequest.getStatus() == Order.Status.ACCEPTED) {
+			this.sendPaymentRequest(orderRequest);
+		}
+
+		//AQUI TRATAMOS O STATUS SCHEDULED QUE VAMOS NA SUPERPAY EFETUAR A RESERVA DO VALOR PARA PAGAMENTO
+		if (orderRequest.getStatus() == Order.Status.SCHEDULED) {
+			this.validateScheduledAndsendPaymentRequest(orderRequest);
+		}
+
 		return orderRepository.save(order);
 	}
 
+	private void validateScheduledAndsendPaymentRequest(Order orderRequest) throws Exception {
+
+		int daysToStart = Integer.parseInt(daysToStartPayment);
+
+		//INSTANCIAMOS O CALENDARIO
+		Calendar c = Calendar.getInstance();
+
+		//DATA ATUAL
+		//EX.: 20/08/2017
+		Date now = c.getTime();
+
+		//PEGAMOS A DATA DE INICIO DO AGENDAMENTO DO PEDIDO
+		Date scheduleDateStart = orderRequest.getScheduleId().getScheduleStart();
+
+		//ATRIBUIMOS A DATA DO AGENDAMENTO DO PEDIDO AO CALENDARIO
+		c.setTime(scheduleDateStart);
+
+		//VOLTAMOS N DIAS, DEFINIDO EM PROPRIEDADES, NO CALENDARIO BASEADO NA DATA DO AGENDAMENTO
+		c.add(Calendar.DATE, -daysToStart);
+
+		//DATA DO AGENDAMENTO MENOS N DIAS NO FORMADO DATE. OU SEJA, A DATA QUE DEVE INICIAR AS TENTAVIDAS DE PAGAMENTO
+		Date dateToStartPayment = c.getTime();
+
+		//SE A DATA ATUAL FOR POSTERIOR A DATA QUE DEVE INICIAR AS TENTATIVAS DE RESERVA DO PAGAMENTO, ENVIAMOS PARA PAGAMENTO
+		if (now.after(dateToStartPayment)) {
+			sendPaymentRequest(orderRequest);
+		}
+	}
+
+	private void sendPaymentRequest(Order orderRequest) throws Exception {
+
+		try {
+			Optional<RetornoTransacao> retornoTransacaoSuperpay = paymentController.sendRequest(orderRequest);
+
+			if(retornoTransacaoSuperpay.isPresent()) {
+
+				//VALIDAMOS SE VEIO O STATUS QUE ESPERAMOS
+				//1 = Pago e Capturado | 2 = Pago e não Capturado (O CORRETO SERIA 1, POIS FAREMOS A CAPTURA POSTERIORMENTE)
```

Comment:
E se por algum descuido da Superpay eles o pedido for capturado tbm? Não estaria errado para o cliente, ele terá feito o pagamento com sucesso.

Acha que deveríamos tratar se veio PAGO E CAPTURADO e mandar cancelar o pagamento para, então, novamente tentar fazer o pagamento do cliente?

A configuração de CAPTURA AUTOMATICA ou MANUAL está configurada na SUPERPAY, se eles mudarem lá para AUTOMATICA, todos os pedidos aprovados virão como PAGO E CAPTURADO e não adianta cancelar e tentar novamente para vir como NÃO CAPTURADO.

E mais, se tentar fazer várias vezes o mesmo pagamento, com o mesmo valor, para o mesmo estabelecimento, geralmente as operadoras de cartão de crédito bloqueiam o cartão por segurança, ou então bloqueiam as vendas seguintes no mesmo estabelecimento.

Creio que o mais prudente seria aceitar o pagamento, internamente notificar para os desenvolvedores e responsáveis que as vendas estão vindo já CAPTURADAS para, então, entrarmos em contato com a SUPERPAY para resolver com eles.

OBS.: ESTOU ACHANDO ESSA SUPERPAY MUITO FUNDO DE QUINTAL, NOS OUTROS GATEWAYS ESSA CAPTURA AUTOMÁTICA OU MANUAL SÃO ENVIADAS POR PARÂMETROS NO MOMENTO DO REQUEST DE CRIAÇÃO DE PEDIDO DE PAGAMENTO.